### DISCUSSION TITLE
Parcel 2 web extension TypeScript validator cannot find name 'browser'

### DISCUSSION THREAD

**Discussion by alpha-tango-kilo** [parcel-bundler/parcel#5897] [Q&A] [answered]
Hi there, I hope this is the right place to ask this question.

I'm using the new Parcel 2 to build a web extension. I'm using [webextension-polyfill-ts](https://github.com/Lusito/webextension-polyfill-ts) to be able to use the MDN-style `browser.` calls in my code while remaining cross-platform.

The issue I've run into is that Parcel's typescript validator (`@parcel/validator-typescript`) doesn't seem to recognise `browser`, despite it being imported in the code using `import { browser } from "webextension-polyfill-ts";`.

I'm not sure if I'm doing something wrong or if this is a bug, so I thought I'd ask here first.

`package.json` dependencies:

```json
"@parcel/config-webextension": "^2.0.0-nightly.2227",
"@parcel/optimizer-cssnano": "^2.0.0-nightly.605",
"@parcel/optimizer-htmlnano": "^2.0.0-nightly.605",
"@parcel/packager-css": "^2.0.0-nightly.605",
"@parcel/packager-html": "^2.0.0-nightly.605",
"@parcel/transformer-css": "^2.0.0-nightly.605",
"@parcel/transformer-html": "^2.0.0-nightly.605",
"@parcel/transformer-postcss": "^2.0.0-nightly.605",
"@parcel/transformer-posthtml": "^2.0.0-nightly.605",
"@parcel/transformer-typescript-tsc": "^2.0.0-nightly.605",
"@parcel/validator-typescript": "^2.0.0-nightly.605",
"parcel": "^2.0.0-nightly.603",
"typescript": "^4.1.5",
"web-ext-types": "^3.2.1",
"webextension-polyfill-ts": "^0.25.0"
```

`.parcelrc`:

```json
{
    "extends": "@parcel/config-webextension",
    "transformers": {
        "*.{ts,tsx}": [
            "@parcel/transformer-typescript-tsc"
        ]
    },
    "validators": {
        "*.{ts,tsx}": [
            "@parcel/validator-typescript"
        ]
    }
}
```

`tsconfig.json`:
```json
{
  "compilerOptions": {
    "typeRoots": ["node_modules/@types", "node_modules/web-ext-types"],
    "outDir": "./dist/",
    "sourceMap": true,
    "strict": true,
    "noImplicitReturns": true,
    "module": "es2015",
    "moduleResolution": "node",
    "target": "es2015",
    "allowJs": true,
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "./src/**/*"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

The project still builds fine, but the output is littered with similar errors to:
```
@parcel/validator-typescript: Cannot find name 'browser'.
  13 |
> 14 |     browser.tabs.onActivated.addListener(onChangeTab);
>    |     ^^^^^^^^ Cannot find name 'browser'.
  15 |
```

How can I resolve this? Happy to provide any further information as needed

**Comment by gregmulvaney** [comment, marked as answer]
I know its a year later but the answer is to install `@types/firefox-webext-browser`

**Reply by alpha-tango-kilo** [reply]
Thanks for replying. I'm well past this project now but I'll mark this as the answer under the presumption it'll work for anyone else with the problem
